databricks/koalas/series.py

```diff
@@ -2803,18 +2783,29 @@ def idxmin(self, skipna=True):
         >>> s.idxmin()
         ('b', 'f')
         """
+        return self._idx(func='idxmin', skipna=skipna)
+
+    def _idx(self, func, skipna):
+        # This is used to idxmax and idxmin
         sdf = self._kdf._sdf
         scol = self._scol
         index_scols = self._kdf._internal.index_scols
         # asc_nulls_(list|first)is used via Py4J directly because
         # it's not supported in Spark 2.3.
-        if skipna:
-            sdf = sdf.orderBy(Column(scol._jc.asc_nulls_last()))
-        else:
-            sdf = sdf.orderBy(Column(scol._jc.asc_nulls_first()))
+        if func == 'idxmin':
+            if skipna:
+                sdf = sdf.orderBy(Column(scol._jc.asc_nulls_last()))
```

Comment:
Hmmmm .. seems like it will execute sort for each column, which will be super expensive .. do you think we can sort once and extract idxmin and idxmax?

Reply:
Do you mean it will sort each column in `frame.idxmin`. I don't have good idea to solve it, because it need get max index in every independent column.

Reply:
hmmmmm .... let's don't add this API for now .. it will be crazy expensive ...

Reply:
Let's leave this closed for now.